src/main/php/web/logging/ToFunction.class.php

```diff
@@ -5,18 +5,18 @@ class ToFunction extends Sink {
 
   /** @param callable $function */
   public function __construct($function) {
-    $this->function= cast($function, 'function(web.Request, web.Response, ?web.Error): void');
+    $this->function= cast($function, 'function(web.Request, web.Response, [:var]): void');
```

Comment:
Technically speaking, this is a BC break. We might want to check and then wrap old functions inside this:

```php
function($req, $res, $hints) use($old) {
  return $old($req, $res, $hints['error'] ?? null);
}
```

Reply:
Merging this with 4.0 as target